test/test_distributions.py

```diff
@@ -2169,6 +2169,94 @@ def test_lazy_probs_initialization(self):
                 self.assertFalse('logits' in vars(dist), msg=message)
 
 
+@unittest.skipIf(not TEST_NUMPY, "NumPy not found")
+class TestAgainstScipy(TestCase):
+    def setUp(self):
+        random_tensor = torch.Tensor([0.7, 0.2, 0.4])
```

Comment:
You have lots of conditionals in the implementations. It might be worth while to use a larger random tensor like
```
random_tensor = torch.Tensor(4, 5).normal_()
```
and then `.exp()` when you need it to be positive etc. Also, it would also be a good idea to define a few of these
```
random_tensor1 = torch.Tensor(4, 5).normal_()
random_tensor2 = torch.Tensor(4, 5).normal_()
random_tensor3 = torch.Tensor(4, 5).normal_()
```
so as to avoid introducing dependence among parameters below (and thereby test only on a hyperplane where things might just happen to work). For example, your tests currently will not detect an accidental swap of `loc`, `scale` because you set them to the same thing :smile:

Reply:
This may be a little tricky the way it's implemented right now, since most scipy distributions don't accept 2d tensors as input. I think I can handle it though.

Reply:
Oh, 2d is not needed, `torch.Tensor(20).normal_()` would work fine. I'm mostly concerned with using independently random tensors for different parameters.